### ISSUE TITLE
Move all Security.php methods to a Trait

### ISSUE THREAD

**Issue by axyr** [CasaLaguna/silverstripe-security#15]
Right now, if you want to use Security features for Frontend Visitors, you have the option to : 

1. Use the Security Controller
2. Build you own if you want to have that same functions in a custom page that lives in the SiteTree

The use of Security.php is pretty limited when you need to use Translatable or if you want to customise aspects of how the controller should work.

If you want to have features like LostPasswordForm, login etc, on a MyPage class, that involves a lot of copying logic from Security.php.

If all the login methods live in a Trait, we could easily drop in the features in a PageType en overload some methods that should behave different.

**Comment by Firesphere** [comment]
Well, that's exactly why the rewrite. And the addition of more update/onAfter/onBefore hooks ;)

I agree a whole lot should/can be moved to a Trait. Not everything though, because Security, in itself, is a controller.
For LostPasswordForm, you can just call the separate LostPasswordForm::create() method, the handling should already work then.

**Comment by axyr** [comment]
Yes, but you can drop in the Trait with all the methods in the Security Controller?
Sure you can copy a lot from Security to your MyCustomLoginPage. but it sill results in a lot of code duplication...
